src/gf.c

```diff
@@ -2032,7 +2075,9 @@ void jl_add_method_to_table(jl_methtable_t *mt, jl_tupletype_t *types, jl_lambda
     jl_sym_t *n = mt->name;
     if (meth->name != anonymous_sym && meth->name != n) {
         // already used by another GF; make a copy (issue #10373)
+        assert(meth->sparam_vals == jl_emptysvec);
         meth = jl_add_static_parameters(meth, jl_emptysvec, NULL);
+        meth->unspecialized = NULL;
```

Comment:
this code will need a new test (the existing test from #10373 no longer invokes it because of the special toplevel closure handling) has already caused the lambda to be newly created..

for example, the following:
```julia
for f in (cos, sin, tan)
    (ft::typeof(f))(x::newtype) = println(ft)
end
```